packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -187,14 +191,34 @@ const App = (props) => {
                 body: {}
             })
         }
+    }, [baskets])
+
+    useEffect(() => {
         // update the basket currency if it doesn't match the current locale currency
-        if (baskets?.baskets?.[0]?.currency && baskets.baskets[0].currency !== currency) {
+        if (basket?.currency && basket?.currency !== currency) {
             updateBasket.mutate({
-                parameters: {basketId: baskets.baskets[0].basketId},
+                parameters: {basketId: basket.basketId},
                 body: {currency}
             })
         }
-    }, [baskets])
+    }, [basket?.currency])
+
+    useEffect(() => {
+        // update the basket customer email
+        if (
+            basket &&
+            customer?.isRegistered &&
+            customer?.email &&
+            customer?.email !== basket?.customerInfo?.email
+        ) {
+            updateCustomerForBasket.mutate({
+                parameters: {basketId: basket.basketId},
+                body: {
+                    email: customer.email
+                }
+            })
+        }
+    }, [customer?.isRegistered, customer?.email, basket?.customerInfo?.email])
```

Comment:
Do we need to track `basket?.customerInfo?.email`? Is there a case where `basket?.customerInfo?.email` is changed manually? 
If `basket?.customerInfo?.email` does change, with the logic we do here, it will soon replaced with whatever is in customer.email. Is that what we want?

Reply:
The `basket?.customerInfo?.email` catches the scenario where we are creating a new basket after an order is submitted.

In this case, the user is already logged in and have just completed checkout so `customer?.isRegistered` and `customer?.email` aren't modified. However, when a new basket is created after the order, it first has `basket?.customerInfo?.email` set to "".  Then since this property changed from the previous basket (which was consumed when the order was submitted) the useEffect triggers, setting the customer's email on the newly created basket.